pkg/retry/unconfirmed_message_handler.go

```diff
@@ -90,26 +84,22 @@ func (s *UnconfirmedMessageHandlerImpl) retryLater() {
 
 // ObserveSending should be called when a new message is sent and it is expected to be [N]ACKed
 func (s *UnconfirmedMessageHandlerImpl) ObserveSending() {
-	s.retryMux.Lock()
-	defer s.retryMux.Unlock()
-	s.numUnackedSendings++
+	s.numUnackedSendings.Add(1)
 	log.Debugf("Observing message being sent. Waiting for an ACK for %s", s.baseInterval.String())
-	if s.numUnackedSendings > 1 {
+	if !s.numUnackedSendings.CompareAndSwap(1, 1) { // if s.numUnackedSendings > 1
```

Comment:
Why use CAS? Wouldn't 
```
if s.numUnackedSendings.Load() > 1 {
```
Yield the same results?

Using CAS here, you're effectively checking if the number is `1` and if it is updating it to `1`.

Reply:
Yeah, that works, thanks!